playbooks/roles/katello_provisioning/tasks/libvirt.yml

```diff
@@ -0,0 +1,57 @@
+---
+- name: 'install the virtualization hypervisor package group'
+  yum:
+    name: "@Virtualization Hypervisor"
+    state: present
+
+- name: 'install the virtualization tools package group'
+  yum:
+    name: "@Virtualization Tools"
+    state: present
+
+- name: 'disable libvirt authentication'
+  lineinfile:
+    dest: /etc/libvirt/libvirtd.conf
+    line: 'auth_unix_rw = "none"'
+
+- name: 'restart libvirt'
+  service: name=libvirtd state=restarted
+
+- name: 'create the provisioning libvirt network'
+  virt_net:
+    command: define
+    name: provision
+    xml: '{{ lookup("template", "../templates/provision.xml.j2") }}'
+
+# Setup TFTP for libvirt
```

Comment:
Other than the setuid I think it should do the steps you mention here. I'd be interested if we need to do more to support this properly.